test/metrics/metrics_test.go

```diff
@@ -0,0 +1,311 @@
+			// given
+			metricsAssertion := InitMetricsAssertion(t, hostAwait, []string{memberAwait.ClusterName, member2Await.ClusterName})
+
+			// when unbaning the user
+			err = hostAwait.Client.Delete(context.TODO(), bannedUser)
+			require.NoError(t, err)
+
+			// then
+			// confirm the BannedUser resource is deleted
+			err = hostAwait.WaitUntilBannedUserDeleted(bannedUser.Name)
+			require.NoError(t, err)
+			// verify the metrics
+			metricsAssertion.WaitForMetricDelta(UserSignupsMetric, 0)         // unchanged: user signup already existed
+			metricsAssertion.WaitForMetricDelta(UserSignupsApprovedMetric, 1) // user approved
+			metricsAssertion.WaitForMetricDelta(UserSignupsBannedMetric, 0)   // unchanged: banneduser already existed
+			metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 1, "domain", "external")
+			metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 1, "cluster_name", memberAwait.ClusterName)  // user provisioned on member1
+			metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", member2Await.ClusterName) // no user on member2
+		})
+	})
+
+	t.Run("manually created usersignup with preexisting banneduser", func(t *testing.T) {
+		// given
+		metricsAssertion := InitMetricsAssertion(t, hostAwait, []string{memberAwait.ClusterName, member2Await.ClusterName})
+		hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().Enabled())
+		id := uuid.Must(uuid.NewV4()).String()
+		email := "testuser" + id + "@test.com"
+		banUser(t, hostAwait, email)
+
+		// when creating the usersignup directly (not via the registration service)
+		userSignup := NewUserSignup(t, hostAwait, "testuser"+id, email)
+		userSignup.Spec.TargetCluster = memberAwait.ClusterName
+		err := hostAwait.CreateWithCleanup(context.TODO(), userSignup)
+		require.NoError(t, err)
+
+		// then
+		// check the UserSignup is created
+		userSignup, err = hostAwait.WaitForUserSignup(userSignup.Name)
+		require.NoError(t, err)
+		// confirm that the user is banned
+		assert.Equal(t, toolchainv1alpha1.UserSignupStateLabelValueBanned, userSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey])
+		mur, err := hostAwait.GetMasterUserRecord(wait.WithMurName("testuser" + id))
+		require.NoError(t, err)
+		assert.Nil(t, mur)
+		// verify the metrics
+		metricsAssertion.WaitForMetricDelta(UserSignupsMetric, 1)
+		metricsAssertion.WaitForMetricDelta(UserSignupsApprovedMetric, 0) // not provisioned because banned before signup
+		metricsAssertion.WaitForMetricDelta(UserSignupsBannedMetric, 1)
+		metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 0, "domain", "external")
+		metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait.ClusterName)  // no user on member1
+		metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", member2Await.ClusterName) // no user on member2
+	})
+
+	t.Run("register new user with preexisting ban", func(t *testing.T) {
```

Comment:
I think that you can drop these two cases - it's a duplication of what is alraedy in the `user_management_test.go` and there is nothing related to metrics that wouldn't be already tested before, right?

Reply:
I don't get it: are we supposed to verify the metrics or not? the goal of this PR is to move all metrics assertions out of the other test cases, so when do you want to check this?

Reply:
>  so when do you want to check this?

check what? what kind of metrics functionality do you test here and wasn't already tested before in `ban provisioned usersignup`?

Reply:
ok, I'll remove this sub-test if you believe it's useless in the scope of metric assertions

Reply:
done in aab86da